src/Blocks/BlockPiston.cpp

```diff
@@ -150,6 +151,12 @@ void cBlockPistonHandler::RetractPiston(Vector3i a_BlockPos, cWorld & a_World)
 				return;
 			}
 
+	if (ExtensionWorldTickTime - World.GetWorldTickAge() <= 2_tick)
+            {
+                // Piston has been activated for less or equal to 2 ticks, bail out
+                return;
+            }
```

Comment:
ExtensionWorldTickTime is not persisted, so when chunks are un- and re-loaded, this will always return when a newly-loaded extended piston is retracted. Is that the correct behaviour?

Reply:
Yeah, this might be an issue if the player let's a 2 tick clock running and goes to unload and load the chunk

Reply:
I'm not sure if this is still relevant with the changes I just pushed